From fa43e2f34706127fea8c763e3b59ac8c97d2796a Mon Sep 17 00:00:00 2001
From: Anton <antone@marvell.com>
Date: Mon, 4 Apr 2016 19:16:41 -0700
Subject: [PATCH 1076/1240] dts: a7040-rz: enable rjmii gop ports in all_pci

The RJMII 1G ports use dedicated PHY_SRD0,1 and
can work with SERDES configured to all PCIe.
SGMII port is disabled as it collides with PCIe0.

Note: mac2 in armada-7040-rz-db_router.dts has been
recently updated to sgmii/2.5G. This did not work
for me, therefore I kept rgmii for both mac2 and mac3.

Change-Id: Ia96e44ec6303d68779f233ab39c5b52073371b3a
Signed-off-by: Anton <antone@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28779
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Yoav Gvili <ygvili@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 arch/arm/dts/armada-7040-rz-db_all_pci.dts | 37 ++++++++++++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/arch/arm/dts/armada-7040-rz-db_all_pci.dts b/arch/arm/dts/armada-7040-rz-db_all_pci.dts
index ae3c45e..61b4165 100644
--- a/arch/arm/dts/armada-7040-rz-db_all_pci.dts
+++ b/arch/arm/dts/armada-7040-rz-db_all_pci.dts
@@ -145,6 +145,43 @@
 					phy-speed = <PHY_SPEED_5G>;
 				};
 			};
+			mdio {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				mac0 {
+					phy-mode = "sgmii"; /* lane-2 */
+					status = "disabled";
+				};
+				mac2 {
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&phy2>;
+					status = "okay";
+				};
+				mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+					status = "okay";
+				};
+			};
+			ppv22@000000 {
+				eth0@010000 {
+					status = "disabled";
+				};
+				eth1@020000 {
+					status = "okay";
+				};
+				eth2@030000 {
+					status = "okay";
+				};
+			};
 		};
 	};
 };
-- 
1.9.1

